File: scripts/register_backend.mjs
#!/usr/bin/env node
/**
 * Auto-register Codespaces backend URL to database.
 * Runs automatically via npm scripts - no manual steps needed.
 */

import { createClient } from '@supabase/supabase-js';

const SUPABASE_URL = process.env.VITE_SUPABASE_URL || 'https://tyxagltygwcvvkcucoeg.supabase.co';
const SUPABASE_KEY = process.env.VITE_SUPABASE_PUBLISHABLE_KEY || 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InR5eGFnbHR5Z3djdnZrY3Vjb2VnIiwicm9sZSI6ImFub24iLCJpYXQiOjE3Njc1NTk3NDcsImV4cCI6MjA4MzEzNTc0N30.XDdmj2yabstWLZ6tilOe1QPNSJ3TSdDH2ZLVj33rud0';

function getCodespacesUrl() {
  const name = process.env.CODESPACE_NAME;
  const domain = process.env.GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN;
  if (name && domain) {
    return `https://${name}-8000.${domain}`;
  }
  return null;
}

async function registerBackend() {
  const url = process.argv[2] || getCodespacesUrl();
  
  if (!url) {
    console.log('⏭️  Not in Codespaces, skipping backend registration');
    return;
  }

  try {
    const supabase = createClient(SUPABASE_URL, SUPABASE_KEY);
    
    const { error } = await supabase
      .from('backend_config')
      .upsert({ id: 'default', api_url: url }, { onConflict: 'id' });

    if (error) {
      console.error('❌ Failed to register backend:', error.message);
      return;
    }

    console.log(`✅ Backend registered: ${url}`);
    console.log('   Lovable preview will auto-discover this URL');
  } catch (err) {
    console.error('❌ Registration error:', err.message);
  }
}

await registerBackend();

File: scripts/register_backend.py
#!/usr/bin/env python3
"""
Register backend URL with Lovable Cloud for frontend discovery.

Usage:
  python scripts/register_backend.py [URL]

If URL is not provided, auto-detects Codespaces URL.
"""
import os
import sys
from supabase import create_client

SUPABASE_URL = os.environ.get("VITE_SUPABASE_URL") or os.environ.get("SUPABASE_URL")
SUPABASE_KEY = os.environ.get("VITE_SUPABASE_PUBLISHABLE_KEY") or os.environ.get("SUPABASE_ANON_KEY")


def get_codespaces_url() -> str | None:
    """Build Codespaces forwarded URL for port 8000."""
    name = os.environ.get("CODESPACE_NAME")
    domain = os.environ.get("GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN")
    if name and domain:
        return f"https://{name}-8000.{domain}"
    return None


def register_url(api_url: str) -> None:
    """Upsert the backend URL into backend_config table."""
    if not SUPABASE_URL or not SUPABASE_KEY:
        print("Error: SUPABASE_URL and SUPABASE_ANON_KEY env vars required")
        print("  Set VITE_SUPABASE_URL and VITE_SUPABASE_PUBLISHABLE_KEY from .env")
        sys.exit(1)

    client = create_client(SUPABASE_URL, SUPABASE_KEY)
    result = client.table("backend_config").upsert({
        "id": "default",
        "api_url": api_url,
    }).execute()

    print(f"✓ Registered backend URL: {api_url}")


def main():
    if len(sys.argv) > 1:
        url = sys.argv[1]
    else:
        url = get_codespaces_url()
        if not url:
            print("Error: Not in Codespaces and no URL provided")
            print("Usage: python scripts/register_backend.py <URL>")
            sys.exit(1)

    register_url(url)


if __name__ == "__main__":
    main()
